apps/asana-service/src/app/api/users/cron/run-users-sync-jobs/route.ts

```diff
@@ -0,0 +1,208 @@
+  let opts = {
+    'workspace': "1205941523188542",
+    'limit': parseInt(process.env.NEXT_PUBLIC_USERS_SYNC_JOB_BATCH_SIZE || '100'),
+    'offset': paginationToken,
+    'opt_fields': "email, name,resource_type, offset, path,uri,workspaces,workspaces.name,workspaces.resource_type, role"
+  };
+
+  try {
+    const result = await usersApiInstance.getUsers(opts);
+
+    if (result && result.data) {
+      return {
+        users: result.data,
+        next_page: result.next_page,
+      };
+    }
+
+    return { users: [], next_page: undefined };
+  } catch (error) {
+    console.error('Error fetching users from Asana:', error);
+    throw new Error('Error fetching users from Asana');
+  }
+}
+
+async function getAccessToken(organisationId: string): Promise<string> {
+
+  const { rows } = await sql`SELECT access_token, refresh_token, expires_in FROM asana_credentials WHERE organisation_id = ${organisationId};`;
+
+  if (rows.length === 0) {
+    throw new Error('No access token is found');
+  }
+
+  let { access_token, refresh_token, expires_in } = rows[0];
+
+  // Check if the current access token is expired
+  if (tokenIsExpired(expires_in)) {
+    const refreshedTokens = await refreshAndSaveAccessToken(organisationId, refresh_token);
+    access_token = refreshedTokens.access_token;
+  }
+
+  return access_token;
+}
+
+async function refreshAndSaveAccessToken(organisationId: string, refreshToken: string): Promise<{ access_token: string, expires_at: number }> {
+  const { access_token, expires_in } = await refreshAccessToken(refreshToken);
+  const expires_at = Math.floor(Date.now() / 1000) + expires_in;
+
+  await sql`
+    UPDATE asana_credentials
+    SET access_token = ${access_token}, expires_in = ${expires_at}
+    WHERE organisation_id = ${organisationId};
+  `;
+
+  return { access_token, expires_at };
+}
+async function refreshAccessToken(refreshToken: string): Promise<RefreshTokenResponse> {
+  // Make a request to the Asana API to refresh the token
+  // Return the new access token, refresh token, and expiry time
+  try {
+    // Ensure environment variables are set
+    if (!process.env.NEXT_PUBLIC_ASANA_CLIENT_ID || !process.env.NEXT_PUBLIC_ASANA_CLIENT_SECRET) {
+      throw new Error('Asana client credentials are not set');
+    }
+
+    const tokenResponse = await axios.post<CodeExchangeData>('https://app.asana.com/-/oauth_token', {
+      grant_type: 'refresh_token',
+      client_id: process.env.NEXT_PUBLIC_ASANA_CLIENT_ID,
+      client_secret: process.env.NEXT_PUBLIC_ASANA_CLIENT_SECRET,
+      refresh_token: refreshToken
+    }, {
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+    });
+  
+    const { access_token, expires_in } = tokenResponse.data;
+    
+    return { access_token, expires_in };
+  } catch (error) {
+    console.error('Error refreshing access token:');
+    throw new Error('Failed to refresh Asana access token'); // Explicitly return undefined
```

Comment:
```suggestion
    throw new Error('Failed to refresh Asana access token', { cause: error }); // Explicitly return undefined
```

Note that, the more information we got in the error we throw, the easier it will be to debug in production.

This comment is applicable to each catch hiding the initial error.